src/query/dblink_scan.c

```diff
@@ -439,13 +605,9 @@ dblink_scan_next (DBLINK_SCAN_INFO * scan_info, val_list_node * val_list)
 SCAN_CODE
 dblink_scan_reset (DBLINK_SCAN_INFO * scan_info)
 {
-  T_CCI_ERROR err_buf;
+  assert (scan_info->conn_handle >= 0 && scan_info->stmt_handle >= 0);
 
-  if (scan_info->conn_handle >= 0 && scan_info->stmt_handle >= 0)
-    {
-      scan_info->cursor = CCI_CURSOR_FIRST;
-      return S_SUCCESS;
-    }
+  scan_info->cursor = CCI_CURSOR_FIRST;
 
-  return S_ERROR;
+  return S_SUCCESS;
```

Comment:
The function always returns S_SUCCESS so how about altering the function return type as void?

Reply:
scan_reset_scan_block calls this routine and it request error code. finally scan_reset_scan_block returns a status (error code)